### ISSUE TITLE
indent back to line start after enter newline and press up

### ISSUE THREAD

**Issue by Yang09701194** [microsoft/vscode#40115]
- VSCode Version: Code 1.18.1 (929bacba01ef658b873545e26034d1a8067445e9, 2017-11-16T18:32:36.026Z)
- OS Version: Windows_NT ia32 10.0.16299

Hi, I have a feature request related to intent,

I think Notepad++ and Visual Studio's indent `after enter newline and press up` works better and more intuitive, it'll remain the indent position, as in the gif:
![vsindent gif](https://user-images.githubusercontent.com/23180780/33894214-5eef9326-df98-11e7-99dc-f97621889720.gif)

But VSCode's indent `after enter newline and press up` go back to line start, which I think is not so reasonable:
![vscodeindent gif](https://user-images.githubusercontent.com/23180780/33894344-b4b70582-df98-11e7-9032-bc7c98a415b6.gif)

Hope vscode can remain the indent position `after enter newline and press up`, thank you.

.

Reproduces without extensions: Yes

**Comment by AdriaandeJongh** [comment]
I desperately want this fixed. Reproduced on macOS with Code 1.24.1 as well.

**Comment by Timofeuz** [comment]
This issue is extremely annoying. Version: 1.27.1 still have it.

**Comment by Icaruk** [comment]
1.28.2 here :(

I didn't had that problem until I formatted my PC, now I have that problem again.
I don't remember how I fixed that before...

**Comment by anavarre** [comment]
Reproduced on 1.29.0 on Linux (KDE Neon)

**Comment by Timofeuz** [comment]
As a partial workaround you can uncheck the setting "Editor: Trim Auto Whitespace". But it still leaves the problem after document being formatted.

**Comment by dropsonic** [comment]
Any updates with this issue?
#45149 that is closed as a duplicate is very annoying. For example, when typing in JavaScript and going to the new line after the curly braces, the cursor is at the beginning of the new line instead of being properly indented.

**Comment by mawyju** [comment]
Bumbing this one up as well.

It's been a few months now that about third of our team switched to VS Code for as3 and JavaScript coding, and this indent-ignoring in class level causes many false-positive conflicts and file change detections on our GIT, which is disturbing especially when merging... Thanks!

**Comment by naresh0839** [comment]
https://github.com/CodeCreator/vscode-indent-to-bracket This may serve as a work around IIUC.

**Comment by dropsonic** [comment]
@naresh0839 thanks for the link. This extension helps, but it doesn't take my indentation settings into account.
I have tabs set up in VS Code but it inserts spaces all over the place.

**Comment by v-kydela** [comment]
@rebornix was assigned to this well over 2 years ago. Why has there been no word from the team?

**Comment by thiagok00** [comment]
this is really annoying, there is any config to fix that? :(

**Comment by v-kydela** [comment]
@ramya-rao-a - Can you comment?

**Comment by desromic** [comment]
Go to File > Preferences > Settings.
Search for "Editor: Trim Auto Whitespace" and disable it. :)

**Comment by dropsonic** [comment]
@desromic well, I'd like to trim auto whitespace at the end of the line, so disabling this option is not suitable

**Comment by longcontest** [comment]
I am using ubuntu and this issue is present in Visual Studio Code here too. I think i want it to be like CLion where indentation is always at the current indent level and not go into inline. Much useful. It must be fixed

**Comment by razzagtaylor** [comment]
Issue still there on Windows

**Comment by lonix1** [comment]
Use [this fork](https://marketplace.visualstudio.com/items?itemName=LAK132.indent-to-bracket) of the "Indent To Bracket" extension which has been more recently maintained.

**Comment by jacquesn** [comment]
The [implicit-indent](https://marketplace.visualstudio.com/items?itemName=jemc.vscode-implicit-indent) extension works for me.

**Comment by JohnCarrera** [comment]
How is this still a problem? I am using VSCode for the first time after working in intellij for years and this is driving me completely nuts. I have to look at the above-mentioned extensions to see if they work for me, thanks everyone!

**Comment by sid-am-ahd935** [comment]
> Go to File > Preferences > Settings. Search for "Editor: Trim Auto Whitespace" and disable it. :)

This worked for me and this was also concerning:


'> @desromic well, I'd like to trim auto whitespace at the end of the line, so disabling this option is not suitable'

So I enabled Files: 
"Trim Trailing White spaces
When enabled, will trim trailing whitespaces when saving the file"

**Comment by briannarenni** [comment]
> > Go to File > Preferences > Settings. Search for "Editor: Trim Auto Whitespace" and disable it. :)
> 
> This worked for me and this was also concerning:
> 
> '> @desromic well, I'd like to trim auto whitespace at the end of the line, so disabling this option is not suitable'
> 
> So I enabled Files: "Trim Trailing White spaces When enabled, will trim trailing whitespaces when saving the file"

I fully suspected that I accidentally changed something at some point because this was a new thing that was on my nerves...that was it! Thanks haha.

**Comment by php4fan** [comment]
How can this be so difficult to fix?? It's been over 6 years!

**Comment by aiday-mar** [comment]
Hi all, apologies for the delay in responding. As some of you have pointed out in this thread, there is already a setting which keeps the trailing whitespaces whenever they are added. The setting name is as follows:

`editor.trimAutoWhitespace`

> @desromic well, I'd like to trim auto whitespace at the end of the line, so disabling this option is not suitable

Is an interesting idea. I will open a separate feature request for this, as I believe the original request on this issue has already been addressed. The community can track the issue there. 

> The only way that setting becomes the full solution to me is if you also add an "on load, assume all blank lines have been auto indented". Then we strip them on save, and create them on load/edit.
> 
> Otherwise I'm still manually triggering auto indent for every first line of new code. I just automatically click on the blank line, between existing code, that is initially shifted all the way to the left (or I don't strip any trailing whitespace).

I am not sure I understand this comment. Could you please elaborate on this @OneOfMany07 ?

In the meantime this issue will be closed as the original feature has been addressed.

**Comment by drankinatty** [comment]
No, No, No. The Auto-trim whitespace setting is NOT a solution to this issue. It is only a workaround if you do NOT want whitespace autotrimmed.

This is still a horribly annoying shortcoming of VSCode which is otherwise and incredibly GREAT IDE. The situation is this. We are talking about specific beginning of empty-line indent that should be left unchanged by Autotrim Whitespace. I still very much want whitespace autotrimmed in the rest of the document.

One way KDE handles this beautifully is to make the Autotrim whitespace active on file-save or file-close. It isn't something that needs to be pedantically done every time I move the cursor of press `[Enter]`. Adding an option `"Editor Trim Auto Whitesace on Save or Close"` can reduce the overhead of the continual checking for autotrim whitespace except on save or file-close. This allows my indent to remain untouched when I press `[Enter]`.

Another alternative would simply be to modify the existing `"Editor Trim Auto Whitesace"` so that it ignore lines with only leading whitespace. That can be a simple regex or pattern match and then just don't remove the initial whitespace.

Implementing either one would be very, very welcomed as this one of the most annoying features of what is otherwise an excellent IDE. (that and not providing a space after the cursor in C/C++ after I press `/* */` for multi-line comment entry `:)`

The automated graphic shown in the first post opening this issue captures the problem exactly and shows how every other editor is able to handle it without frustration. Here is to hope this very old issue gets fixed.

**Comment by aiday-mar** [comment]
Hi @drankinatty yes I understood the feature request, hence I opened the following issue (https://github.com/microsoft/vscode/issues/209939). You may track the feature there.

**Comment by drankinatty** [comment]
Thank you, thank you! I'll follow along there `:)`

**Comment by classiebit** [comment]
"notebook.insertFinalNewline": false
